File: mobile/silvicola_app/lib/domain/entities/arbol.dart
import 'package:equatable/equatable.dart';

class Arbol extends Equatable {
  final int? id;
  final String codigo;
  final int especieId;
  final int parcelaId;
  final double dap;
  final double altura;
  final double latitud;
  final double longitud;
  final String? fotoPath;
  final String? observaciones;
  final String estado;
  final DateTime fechaMedicion;
  final bool sincronizado;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Arbol({
    this.id,
    required this.codigo,
    required this.especieId,
    required this.parcelaId,
    required this.dap,
    required this.altura,
    required this.latitud,
    required this.longitud,
    this.fotoPath,
    this.observaciones,
    required this.estado,
    required this.fechaMedicion,
    this.sincronizado = false,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        codigo,
        especieId,
        parcelaId,
        dap,
        altura,
        latitud,
        longitud,
        fotoPath,
        observaciones,
        estado,
        fechaMedicion,
        sincronizado,
        createdAt,
        updatedAt,
      ];

  Arbol copyWith({
    int? id,
    String? codigo,
    int? especieId,
    int? parcelaId,
    double? dap,
    double? altura,
    double? latitud,
    double? longitud,
    String? fotoPath,
    String? observaciones,
    String? estado,
    DateTime? fechaMedicion,
    bool? sincronizado,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Arbol(
      id: id ?? this.id,
      codigo: codigo ?? this.codigo,
      especieId: especieId ?? this.especieId,
      parcelaId: parcelaId ?? this.parcelaId,
      dap: dap ?? this.dap,
      altura: altura ?? this.altura,
      latitud: latitud ?? this.latitud,
      longitud: longitud ?? this.longitud,
      fotoPath: fotoPath ?? this.fotoPath,
      observaciones: observaciones ?? this.observaciones,
      estado: estado ?? this.estado,
      fechaMedicion: fechaMedicion ?? this.fechaMedicion,
      sincronizado: sincronizado ?? this.sincronizado,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: backend/ForestInventory/src/ForestInventory.Domain/Entities/Arbol.cs
using ForestInventory.Domain.Enums;

namespace ForestInventory.Domain.Entities;

public class Arbol
{
    public Guid Id { get; set; }
    public string Codigo { get; set; } = string.Empty;
    public int NumeroArbol { get; set; } // Número secuencial del árbol dentro de la parcela
    public double Latitud { get; set; }
    public double Longitud { get; set; }
    public double? Altitud { get; set; }
    public double Dap { get; set; } // Diámetro a la altura del pecho (cm)
    public double Altura { get; set; } // Altura total (m)
    public double? AlturaComercial { get; set; } // Altura comercial (m)
    public double? DiametroCopa { get; set; } // Diámetro de copa (m)
    public EstadoArbol Estado { get; set; }
    public string? Observaciones { get; set; }
    public DateTime FechaMedicion { get; set; }
    public DateTime FechaCreacion { get; set; }
    public DateTime? FechaUltimaActualizacion { get; set; }
    public bool Sincronizado { get; set; }
    public Guid? SyncId { get; set; }

    // Foreign Keys
    public Guid ParcelaId { get; set; }
    public Guid EspecieId { get; set; }
    public Guid UsuarioCreadorId { get; set; }

    // Navigation properties
    public Parcela Parcela { get; set; } = null!;
    public Especie Especie { get; set; } = null!;
    public Usuario UsuarioCreador { get; set; } = null!;

    // Calculated properties
    public double CalcularAreaBasal()
    {
        return Math.PI * Math.Pow(Dap / 2, 2) / 10000;
    }

    public double CalcularVolumen()
    {
        return CalcularAreaBasal() * Altura * 0.7;
    }
}
